diff --git a/arch/arm/boot/dts/stm32mp13-pinctrl.dtsi b/arch/arm/boot/dts/stm32mp13-pinctrl.dtsi
index 6a7ae1447715..8fe4084a090f 100644
--- a/arch/arm/boot/dts/stm32mp13-pinctrl.dtsi
+++ b/arch/arm/boot/dts/stm32mp13-pinctrl.dtsi
@@ -180,6 +180,16 @@ pins {
 		};
 	};
 
+	i2c4_pins_a: i2c4-0 {
+		pins {
+			pinmux = <STM32_PINMUX('E', 15, AF6)>, /* I2C4_SCL */
+				 <STM32_PINMUX('B', 9, AF6)>; /* I2C4_SDA */
+			bias-disable;
+			drive-open-drain;
+			slew-rate = <0>;
+		};
+	};
+
 	i2c5_pins_a: i2c5-0 {
 		pins {
 			pinmux = <STM32_PINMUX('D', 1, AF4)>, /* I2C5_SCL */
diff --git a/drivers/mfd/stpmic1.c b/drivers/mfd/stpmic1.c
index 7ff37a742779..8ce206ea4aaf 100644
--- a/drivers/mfd/stpmic1.c
+++ b/drivers/mfd/stpmic1.c
@@ -138,10 +138,11 @@ static int stpmic1_probe(struct i2c_client *i2c,
 		return PTR_ERR(ddata->regmap);
 
 	ddata->irq = of_irq_get(np, STPMIC1_MAIN_IRQ);
-	if (ddata->irq < 0) {
-		dev_err(dev, "Failed to get main IRQ: %d\n", ddata->irq);
-		return ddata->irq;
-	}
+        //TODO
+	//if (ddata->irq < 0) {
+	//	dev_err(dev, "Failed to get main IRQ: %d\n", ddata->irq);
+	//	return ddata->irq;
+	//}
 
 	ret = regmap_read(ddata->regmap, VERSION_SR, &reg);
 	if (ret) {
@@ -150,15 +151,16 @@ static int stpmic1_probe(struct i2c_client *i2c,
 	}
 	dev_info(dev, "PMIC Chip Version: 0x%x\n", reg);
 
-	/* Initialize PMIC IRQ Chip & associated IRQ domains */
-	ret = devm_regmap_add_irq_chip(dev, ddata->regmap, ddata->irq,
-				       IRQF_ONESHOT | IRQF_SHARED,
-				       0, &stpmic1_regmap_irq_chip,
-				       &ddata->irq_data);
-	if (ret) {
-		dev_err(dev, "IRQ Chip registration failed: %d\n", ret);
-		return ret;
-	}
+        //TODO
+	///* Initialize PMIC IRQ Chip & associated IRQ domains */
+	//ret = devm_regmap_add_irq_chip(dev, ddata->regmap, ddata->irq,
+	//			       IRQF_ONESHOT | IRQF_SHARED,
+	//			       0, &stpmic1_regmap_irq_chip,
+	//			       &ddata->irq_data);
+	//if (ret) {
+	//	dev_err(dev, "IRQ Chip registration failed: %d\n", ret);
+	//	return ret;
+	//}
 
 	return devm_of_platform_populate(dev);
 }
diff --git a/drivers/regulator/stpmic1_regulator.c b/drivers/regulator/stpmic1_regulator.c
index ff397c5eba5a..1d3b8c596189 100644
--- a/drivers/regulator/stpmic1_regulator.c
+++ b/drivers/regulator/stpmic1_regulator.c
@@ -360,7 +360,7 @@ static const struct regulator_ops stpmic1_switch_regul_ops = {
 	.type = REGULATOR_VOLTAGE, \
 	.owner = THIS_MODULE, \
 	.min_uV = 0, \
-	.fixed_uV = 5000000, \
+	.fixed_uV = 3300000, \
 	.enable_reg = BST_SW_CR, \
 	.enable_mask = SWIN_SWOUT_ENABLED, \
 	.enable_val = SWIN_SWOUT_ENABLED, \
